dask_kubernetes/operator/kubecluster/kubecluster.py

```diff
@@ -543,13 +552,20 @@ async def _close(self, timeout=None):
         if self.shutdown_on_close:
             async with kubernetes.client.api_client.ApiClient() as api_client:
                 custom_objects_api = kubernetes.client.CustomObjectsApi(api_client)
-                await custom_objects_api.delete_namespaced_custom_object(
-                    group="kubernetes.dask.org",
-                    version="v1",
-                    plural="daskclusters",
-                    namespace=self.namespace,
-                    name=self.name,
-                )
+                try:
+                    await custom_objects_api.delete_namespaced_custom_object(
+                        group="kubernetes.dask.org",
+                        version="v1",
+                        plural="daskclusters",
+                        namespace=self.namespace,
+                        name=self.name,
+                    )
+                except ApiException as e:
+                    if e.reason == "Not Found":
+                        # TODO: maybe add a warning or do nothing?
```

Comment:
Let's add a warning here. Something like `Failed to delete DaskCluster, looks like it has already been deleted.`.